mbed_project/_internal/git_utils.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Wrappers for git operations."""
+from dataclasses import dataclass
+from pathlib import Path
+
+import git
+
+from mbed_project.exceptions import VersionControlError
+from mbed_project._internal.progress import ProgressReporter
+
+
+@dataclass
+class GitReference:
+    """Git reference for a remote repository.
+
+    Attributes:
+        repo_url: URL of the git repository.
+        ref: The reference commit sha, tag or branch.
+    """
+
+    repo_url: str
+    ref: str
+
+
+def clone(url: str, dst_dir: Path) -> git.Repo:
+    """Clone a library repository.
+
+    Args:
+        git_ref: Object containing git information.
```

Comment:
this is not matching the arguments

Reply:
Nice catch, this docstring is from an earlier revision of the code and I forgot to change it!

Reply:
c9ec845841bf1c77a0bcf59950d825f34e3a1d78